src/libAtomVM/bif.c

```diff
@@ -197,6 +197,21 @@ term bif_erlang_is_tuple_1(Context *ctx, uint32_t fail_label, term arg1)
     return term_is_tuple(arg1) ? TRUE_ATOM : FALSE_ATOM;
 }
 
+term bif_erlang_is_record_2(Context *ctx, uint32_t fail_label, term arg1, term arg2)
+{
+    UNUSED(ctx);
+    UNUSED(fail_label);
+
+    VALIDATE_VALUE_BIF(fail_label, arg1, term_is_tuple);
+    VALIDATE_VALUE_BIF(fail_label, arg2, term_is_atom);
+    if (term_get_tuple_arity(arg1) < 1) {
+        RAISE_ERROR_BIF(fail_label, BADARG_ATOM);
+    }
+
+    term tag = term_get_tuple_element(arg1, 0);
+    return tag == arg2 ? TRUE_ATOM : FALSE_ATOM;
```

Comment:
Not sure if we also want to have `VALIDATE_VALUE_BIF` after tag extraction or returning false is enough. `erl` returns false:

```
3> -record(person, {id, name, age}).
ok
4> erlang:is_record({1,2,3}, person).
false
```

Reply:
We want the same behavior as BEAM and this can be verified with a test that is run on both BEAM and AtomVM.